route-rs-runtime/src/utils/test/harness.rs

```diff
@@ -76,8 +31,7 @@ pub fn initialize_runtime() -> runtime::Runtime {
         .unwrap()
 }
 
-// This is kind of ugly ish, right now
-pub async fn execute_link<OutputPacket: Debug + Send + Clone + 'static>(
+pub async fn run_link<OutputPacket: Debug + Send + Clone + 'static>(
```

Comment:
Initializing tests in the async runtime context is fine for now. It might be interesting to see if we could get a `#[with_tokio_runtime]` macro working so we could annotate `async fn`s. That would be a nice easy win to reduce boilerplate and 1 level of indentation.

Reply:
Yeah, macros are kind of scary but yeah we should try to figure out some sort of macro. Will think about this for a future improvement.

Reply:
Small issue is that the test fn's can not be async. Which is annoying, but we may be able to work around.

Reply:
That's so annoying! There will always be more "make testing better" work todo.